pkgs/sketch_pad/lib/editor/editor.dart

```diff
@@ -173,6 +173,21 @@ class _EditorWidgetState extends State<EditorWidget> implements EditorService {
         (CodeMirror editor, [HintOptions? options]) {
           return options!.results;
         }.toJS);
+
+    // Listen for document body to be visible, then force a code mirror refresh.
+    final observer = web.IntersectionObserver(
+      (JSArray entries, web.IntersectionObserver observer) {
+        for (final entry in entries.toDart) {
+          if ((entry as web.IntersectionObserverEntry).isIntersecting) {
+            observer.unobserve(web.document.body!);
+            Timer.run(() => codeMirror!.refresh());
+            return;
+          }
+        }
+      }.toJS,
+    );
+
+    observer.observe(web.document.body!);
```

Comment:
@parlough I don't think this is what you want, if you observe `body`, `isIntersecting` will always be `true` (the body is always intersecting with itself!). I think what you want to observe is the container of the codeMirror editor.

The current implementation is effectively calling `Timer.run` on "the next frame" after you're told that the DIV is rendered in every codeMirror editor in the page.

If you want to use the intersection observer (fine by me, that's what it's for!) you need to tweak this to something like this JS-Fiddle demo I've prepared here, check it out:

* https://jsfiddle.net/ditman/x93b7gj8/

(Your code is 99% there)

Reply:
Thanks for looking into this @ditman!

I guess I was only thinking about the case where DartPad was embedded in an iframe and that document's body being visible within the user's view port. Or does `IntersectionObserver` not consider the view port of the parent document?

Either way, observing the container of the editor makes a lot more sense, and it sounds like it would account for other issues where the embed is in the viewport, but not the editor, especially for the mobile layout. I'll try that out and open a new PR. Thanks!

Reply:
> Or does IntersectionObserver not consider the view port of the parent document?

I don't think it does, but I've never tried that level of nesting! In that case, you might be able to simplify your solution to just schedule the `reset` of the editor directly (on a `Timer` or a `Future.delayed`, whatever you prefer).

> I'll try that out and open a new PR.

Feel free to ping me if any of the changes you want to make to the embedding gives you any grief, we can always go to the basic JS and then translate the fix to package:web!